File: leetcode/tree.py
from collections import deque


# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


def String2Tree(s: str) -> TreeNode:
    vals = s.split()
    if len(vals) == 0:
        return None
    root = TreeNode(int(vals[0]))
    q = deque()
    q.append(root)
    idx = 1
    while len(q) > 0:
        node = q.popleft()
        left_idx = idx
        if left_idx < len(vals) and vals[left_idx] != '#':
            node.left = TreeNode(int(vals[left_idx]))
            q.append(node.left)
        right_idx = idx + 1
        if right_idx < len(vals) and vals[right_idx] != '#':
            node.right = TreeNode(int(vals[right_idx]))
            q.append(node.right)
        idx += 2
    return root

File: leetcode/tree.rs
use std::collections::VecDeque;
use std::rc::Rc;
use std::cell::{Ref, RefCell};

// Definition for a binary tree node.
#[derive(Debug, PartialEq, Eq)]
pub struct TreeNode {
    pub val: i32,
    pub left: Option<Rc<RefCell<TreeNode>>>,
    pub right: Option<Rc<RefCell<TreeNode>>>,
}

impl TreeNode {
    #[inline]
    pub fn new(val: i32) -> Self {
        TreeNode {
            val,
            left: None,
            right: None
        }
    }
}


pub fn get_node_ref(node: &Option<Rc<RefCell<TreeNode>>>) -> Ref<TreeNode> {
  return node.as_ref().unwrap().borrow();
}

pub fn create_treenode(s: &str) -> Option<Rc<RefCell<TreeNode>>> {
    let mut queue: VecDeque<Rc<RefCell<TreeNode>>> = VecDeque::new();
    let mut iter = s.split_ascii_whitespace();
    let val = iter.next();
    if val.is_none() {
        return None;
    }
    let num = val.unwrap().to_string().parse::<i32>().unwrap();
    let root = Rc::new(RefCell::new(TreeNode::new(num)));
    queue.push_back(Rc::clone(&root));
    while queue.len() > 0 {
        let parent = queue.pop_front().unwrap();
        let val = iter.next();
        if val.is_some() && val.unwrap() != "#" {
            let num = val.unwrap().to_string().parse::<i32>().unwrap();
            let node = Rc::new(RefCell::new(TreeNode::new(num)));
            parent.borrow_mut().left = Some(Rc::clone(&node));
            queue.push_back(Rc::clone(&node));
        }
        let val = iter.next();
        if val.is_some() && val.unwrap() != "#" {
            let num = val.unwrap().to_string().parse::<i32>().unwrap();
            let node = Rc::new(RefCell::new(TreeNode::new(num)));
            parent.borrow_mut().right = Some(Rc::clone(&node));
            queue.push_back(Rc::clone(&node));
        }
    }
    return Some(root);
}
